### PR TITLE
[EGD-8095] show quote on wallpaper

### PR DESCRIPTION
**Pull request by wojtekrzepecki**

### CODE REVIEW

**Comment by pholat** [comment]
please remove images and add it to MuditaOSPublicAssets + add new image release :) 
image

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 namespace gui
 {
     PhoneLockedWindow::PhoneLockedWindow(app::ApplicationCommon *app, const std::string &name)
         : AppWindow(app, name),
           notificationsModel(std::make_shared<NotificationsModel>(NotificationsListPlacement::LockedScreen))
     {
+        clockWallpaper      = std::make_shared<WallpaperClock>(this, notificationsModel);
+        auto quoteWallpaper = std::make_shared<WallpaperQuote>(app, this);
+        auto logoWallpaper  = std::make_shared<WallpaperLogo>(this);
+        wallpaperPresenter  = std::make_unique<WallpaperPresenter>(app, clockWallpaper, quoteWallpaper, logoWallpaper);
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp:20]: please pass presenter from application :)

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp [hunk 2]
```diff
@@ -80,6 +61,13 @@ namespace gui
         }
     }
 
+    bool PhoneLockedWindow::updateTime()
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp:64]: consider avoiding order changes

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp [hunk 3]
```diff
@@ -17,6 +19,10 @@ namespace gui
 {
     class PhoneLockedWindow : public AppWindow
     {
+      private:
+        std::shared_ptr<WallpaperClock> clockWallpaper         = nullptr;
+        std::unique_ptr<WallpaperPresenter> wallpaperPresenter = nullptr;
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.hpp:24]: pass a generic/non generic presenter from the application and consider using reference then

File: module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperBase.hpp [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    class WallpaperBase
+    {
+      protected:
+        Item *parent = nullptr;
+
+      public:
+        explicit WallpaperBase(Item *parent);
+        virtual void build() = 0;
+        virtual void show()  = 0;
+        virtual void hide()  = 0;
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperBase.hpp:19]: why do we shod/hide them - not add/remove them in the place required instead?

File: module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperQuote.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+#include "WallpaperQuote.hpp"
+#include <service-appmgr/Controller.hpp>
+#include <ImageBox.hpp>
+#include <Text.hpp>
+#include <Style.hpp>
+
+namespace gui
+{
+    WallpaperQuote::WallpaperQuote(app::ApplicationCommon *app, Item *parent)
+        : WallpaperBase(parent), quotesPresenter{std::make_unique<QuotesPresenter>(app)}
```
Comment (pholat) [inline module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperQuote.cpp:13]: :+1:

File: module-apps/apps-common/popups/presenter/WallpaperPresenter.cpp [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+                break;
+            case WallpaperOption::Quote:
+                clockWallpaper->hide();
+                quoteWallpaper->show();
+                logoWallpaper->hide();
+                break;
+            case WallpaperOption::Logo:
+                clockWallpaper->hide();
+                quoteWallpaper->hide();
+                logoWallpaper->show();
```
Comment (pholat) [inline module-apps/apps-common/popups/presenter/WallpaperPresenter.cpp:40]: consider: `'view->setWalpaper(new Wallpaper(option))`

**Review by pholat — COMMENTED** [review]
seems legit except a few minor change requests

**Review by PrzeBrudny — APPROVED** [review]
Ok -> but please apply @pholat suggestions (I agree with them).

File: module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperBase.hpp (hunk 4 above)
Comment (wojtekrzepecki) [reply module-apps/apps-common/popups/lock-popups/phone-lock-wallpapers/WallpaperBase.hpp:19]: All elements have to exist - currrent notification models requires it

File: module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp (hunk 1 above)
Comment (wojtekrzepecki) [reply module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp:20]: Presenter needs wallpapers (safest to force passing them through constructor) and wallpapers need window item to be cnostructed -> thus this forces current order of initialization.

Comment (pholat) [reply module-apps/apps-common/popups/lock-popups/PhoneLockedWindow.cpp:20]: `presenter needs wallpapers` tbh - it needs view which it triggers it shall not bother how views are implemented. 
It's just the wrong place to connect the presenter to view. Code here is just an aftermath of lack of: https://github.com/mudita/MuditaOS/pull/3463/files#r781229360

**Review by pholat — APPROVED** [review]

**Pull request merged**